File: backend/internal/types/user_list.go
package types

type UserList struct {
	ID       int `json:"id"`
	ListID   int `json:"listID"`
	MemberID int `json:"memberID"`
}

type EntireUserList struct {
	ID        int   `json:"id"`
	ListID    int   `json:"listID"`
	MemberIDs []int `json:"memberIDs"`
}

File: app/lib/hive/user_list.dart
import 'package:hive/hive.dart';

part 'user_list.g.dart';

@HiveType(typeId: 11)
class UserList extends HiveObject {
  @HiveField(0)
  int id;

  @HiveField(1)
  int listID;

  @HiveField(2)
  List<int> memberIDs;

  UserList({
    required this.id,
    required this.listID,
    required this.memberIDs,
  });
}
